api/solarperformanceinsight_api/compute.py

```diff
@@ -444,3 +507,293 @@ def compare_expected_and_actual(job: models.StoredJob, si: storage.StorageInterf
+        ),
+    )
+    ref_pdc_gen = generate_job_performance_data(
+        job,
+        si,
+        types=[models.JobDataTypeEnum.predicted_performance_dc],
+        performance_granularity=(
+            job_params.predicted_data_parameters.performance_granularity
+        ),
+    )
+    results_list = []
+    # Loop through at the inverter level
+    for i, (chain, ref_weather, actual_weather, ref_pac, ref_pdc,) in enumerate(
+        zip_longest(
+            chains,
+            ref_weather_gen,
+            actual_weather_gen,
+            ref_pac_gen,
+            ref_pdc_gen,
+        )
+    ):
+        # use the ModelChain for converting any temperature to cell temperature and
+        # converting irradiance to POA. In the future, could only run these conversions
+        # instead of full modelchain calculation.
+        # since it sets class properties, copy for the actuals calculations
+        chain_actual = deepcopy(chain)
+        inv = job.definition.system_definition.inverters[i]
+        pac0 = inv.inverter_parameters._pac0
+        num_arrays = len(chain.system.arrays)
+        gammas = [
+            arr._gamma for arr in job.definition.system_definition.inverters[i].arrays
+        ]
+        if any([g is None for g in gammas]):
+            raise TypeError(
+                "Currently unable to compare predicted and actual performance for "
+                "PVsyst specified arrays."
+            )
+        if data_available == models.PredictedDataEnum.weather_only:  # 2A-4
+            db_results, _ = process_single_modelchain(
+                chain, ref_weather, ref_model_method, tshift, i
+            )
+            results_list += db_results
+            ref_pdc = adjust(_get_mc_dc(chain.results, num_arrays))  # type: ignore
+        else:
+            # run chain on ref weather
+            getattr(chain, ref_model_method)(
+                [d.shift(freq=tshift) for d in ref_weather]  # type: ignore
+            )
+
+        # run chain on actual weather
+        getattr(chain_actual, actual_model_method)(
+            [d.shift(freq=tshift) for d in actual_weather]  # type: ignore
+        )
+        # use pvlib.modelchain._irrad_for_celltemp that returns POA global if available
+        # otherwise uses effective irradiance
+        poa_ref = _irrad_for_celltemp(
+            chain.results.total_irrad, chain.results.effective_irradiance
+        )
+        poa_actual = _irrad_for_celltemp(
+            chain_actual.results.total_irrad, chain_actual.results.effective_irradiance
+        )
+        # use cell temperature from the pvlib modelchain
+        # If air temp + wind speed were supplied, they are converted
+        # to cell temperature via the array's temperature model.  If
+        # module_temperature was supplied and the temperature model is
+        # sapm, it is converted to cell_temperature, otherwise module
+        # temperature is used in place of cell_temperature
```

Comment:
No. If you just give `ModelChain` module_temperature and the temperature model is not SAPM, it runs a default `air_temp = 20` and `wind_speed = 0` through the temperature model to calculate cell_temperature

Reply:
It the model is SAPM, it does appropriately convert from module to cell temp

Reply:
ugh https://github.com/pvlib/pvlib-python/issues/1113

Reply:
@wholmgren would this SPI issue be fixed by [pvlib/pvlib-python#1113](https://github.com/pvlib/pvlib-python/issues/1113)?

Reply:
Well, it depends on what we mean by made worse. Removing the defaults would mean that the SPI code would break unless we handled the cases properly. But explicitly handling those cases would have prevented my confusion.